### PR TITLE
Fyle Card <> Vendor Mapping setup

### PR DESCRIPTION
**Pull request by Hrishabh17**

### CODE REVIEW

File: apps/workspaces/serializers.py [hunk 1]
```diff
@@ -158,6 +158,17 @@ def create(self, validated_data):
         # Update workspace onboarding state
         workspace = export_settings.workspace
 
+        if export_settings.credit_card_expense_export_type == 'PURCHASE_INVOICE':
```
[the remark this answers is not in the document]
Comment (Hrishabh17) [reply apps/workspaces/serializers.py:161]: yes we are defaulting, no need of toggle.

File: scripts/002_add_vendor_fyle_card_mapping.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+where not exists (
+    select 1
+    from mapping_settings ms 
+    inner join export_settings es 
+    on es.workspace_id = ms.workspace_id
+    where ms.source_field = 'CORPORATE_CARD'
+    and ms.destination_field = 'VENDOR'
+    and ms.workspace_id = ws.id
+    and es.credit_card_expense_export_type = 'PURCHASE_INVOICE'
+);
```
Comment (ashwin1111) [inline scripts/002_add_vendor_fyle_card_mapping.sql:16]: add count(*) from mapping_settings for new insertion and count(*) from export where ccc is PI - just for manual validation

File: tests/test_sage300/test_exports/test_base_model.py [hunk 3]
```diff
@@ -220,6 +224,59 @@ def test_get_vendor_id_3(
...
+    accounting_export.save()
+
+    return_value = base_model.get_vendor_id(
+        accounting_export=accounting_export
+    )
+
+    assert return_value is not None
+    assert return_value == vendor.destination_id
+
+
```
Comment (ashwin1111) [inline tests/test_sage300/test_exports/test_base_model.py:279]: add 1 additional test by having 1 without corporate_card_id, 1 with cc id X, 1 with cc id Y - there should be 3 groups

**Review by ashwin1111 — APPROVED** [review]

File: scripts/002_add_vendor_fyle_card_mapping.sql [hunk 4]
```diff
@@ -1,6 +1,10 @@
 rollback;
 begin;
 
+select COUNT(*) from export_settings where credit_card_expense_export_type = 'PURCHASE_INVOICE';
+
+select COUNT(distinct(workspace_id)) from mapping_settings where workspace_id  not in (select distinct(workspace_id) from mapping_settings where source_field = 'CORPORATE_CARD' and destination_field = 'VENDOR');
```
Comment (ashwin1111) [inline scripts/002_add_vendor_fyle_card_mapping.sql:6]: do both post insert

File: tests/test_sage300/test_exports/test_base_model.py [hunk 5]
```diff
@@ -277,6 +277,92 @@ def test_get_vendor_id_4(
...
+    export_settings = ExportSetting.objects.filter(workspace_id=workspace_id).first()
+    export_settings.credit_card_expense_grouped_by = 'REPORT'
+    export_settings.credit_card_expense_date = 'POSTED_AT'
+    export_settings.reimbursable_expense_grouped_by = 'REPORT'
+    export_settings.reimbursable_expense_date = 'PAYMENT_PROCESSING'
+    export_settings.save()
+
+    accounting_export = _group_expenses(expenses=expenses, export_setting=export_settings, fund_source='CCC')
+    assert len(accounting_export) == 3
```
Comment (ashwin1111) [inline tests/test_sage300/test_exports/test_base_model.py:364]: noice!

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**